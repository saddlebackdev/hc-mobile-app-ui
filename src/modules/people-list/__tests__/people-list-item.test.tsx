import * as React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

import UserItem from '../people-list-item';

const onPressMock = jest.fn();

describe('UserItem', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <UserItem
            name={'Stephen John'}
            profilePic={'https://www.google.com'}
            userId={'512345'}
          />
        </ThemeProvider>,
      );

      const headerWrapper = wrapper.queryByTestId('main-wrapper');

      expect(headerWrapper).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('hides icon when it is null', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <UserItem name={'Stephen John'} profilePic={null} userId={'512345'} />
        </ThemeProvider>,
      );

      const icon = wrapper.queryByTestId('profile-pic');

      expect(icon).toBeNull();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call the onPress method when card is pressed', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <UserItem
            name={'Stephen John'}
            profilePic={null}
            userId={'512345'}
            onPress={onPressMock}
          />
        </ThemeProvider>,
      );

      const linkWrapper = wrapper.getByTestId('link-wrapper');

      expect(linkWrapper).toBeDefined();

      fireEvent.press(linkWrapper);

      expect(onPressMock).toHaveBeenCalledTimes(1);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
