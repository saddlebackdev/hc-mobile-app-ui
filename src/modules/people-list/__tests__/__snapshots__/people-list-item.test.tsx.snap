// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Interactions should call the onPress method when card is pressed 1`] = `
<View
  leftActionActivationDistance={0}
  leftActionReleaseAnimationConfig={null}
  leftActionReleaseAnimationFn={null}
  leftButtonWidth={85}
  leftButtonsActivationDistance={75}
  leftButtonsCloseReleaseAnimationConfig={null}
  leftButtonsCloseReleaseAnimationFn={null}
  leftButtonsOpenReleaseAnimationConfig={null}
  leftButtonsOpenReleaseAnimationFn={null}
  onLayout={[Function]}
  onLeftActionActivate={[Function]}
  onLeftActionComplete={[Function]}
  onLeftActionDeactivate={[Function]}
  onLeftActionRelease={[Function]}
  onLeftButtonsActivate={[Function]}
  onLeftButtonsCloseComplete={[Function]}
  onLeftButtonsCloseRelease={[Function]}
  onLeftButtonsDeactivate={[Function]}
  onLeftButtonsOpenComplete={[Function]}
  onLeftButtonsOpenRelease={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onPanAnimatedValueRef={[Function]}
  onRef={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onRightActionActivate={[Function]}
  onRightActionComplete={[Function]}
  onRightActionDeactivate={[Function]}
  onRightActionRelease={[Function]}
  onRightButtonsActivate={[Function]}
  onRightButtonsCloseComplete={[Function]}
  onRightButtonsCloseRelease={[Function]}
  onRightButtonsDeactivate={[Function]}
  onRightButtonsOpenComplete={[Function]}
  onRightButtonsOpenRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeComplete={[Function]}
  onSwipeMove={[Function]}
  onSwipeRelease={[Function]}
  onSwipeStart={[Function]}
  rightActionActivationDistance={125}
  rightActionReleaseAnimationConfig={null}
  rightActionReleaseAnimationFn={null}
  rightButtonWidth={85}
  rightButtonsActivationDistance={75}
  rightButtonsCloseReleaseAnimationConfig={null}
  rightButtonsCloseReleaseAnimationFn={null}
  rightButtonsOpenReleaseAnimationConfig={null}
  rightButtonsOpenReleaseAnimationFn={null}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  swipeReleaseAnimationConfig={
    Object {
      "duration": 250,
      "easing": [Function],
      "toValue": Object {
        "x": 0,
        "y": 0,
      },
    }
  }
  swipeReleaseAnimationFn={[Function]}
  swipeStartMinDistance={15}
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="link-wrapper"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "paddingLeft": 22,
              "paddingRight": 11,
              "paddingVertical": 11,
            },
          ]
        }
        testID="main-wrapper"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "width": 40,
              }
            }
            testID="avatar-button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EDF1F5",
                    "borderColor": "#DBE0E3",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1C2530",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "600",
                      "textAlign": "left",
                    },
                  ]
                }
                testID="text"
                weight="semiBold"
              >
                SJ
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingLeft": 11,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1C2530",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "textAlign": "left",
                    },
                  ]
                }
                testID="heading"
                variant="h4"
              >
                Stephen John
              </Text>
            </View>
            <Text
              alignment="right"
              muted={true}
              style={
                Array [
                  Object {
                    "color": "#97A4AB",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "textAlign": "right",
                  },
                ]
              }
              testID="user-id"
              variant="subtitle2"
            >
              ID:512345
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`UserItem Rendering hides icon when it is null 1`] = `
<View
  leftActionActivationDistance={0}
  leftActionReleaseAnimationConfig={null}
  leftActionReleaseAnimationFn={null}
  leftButtonWidth={85}
  leftButtonsActivationDistance={75}
  leftButtonsCloseReleaseAnimationConfig={null}
  leftButtonsCloseReleaseAnimationFn={null}
  leftButtonsOpenReleaseAnimationConfig={null}
  leftButtonsOpenReleaseAnimationFn={null}
  onLayout={[Function]}
  onLeftActionActivate={[Function]}
  onLeftActionComplete={[Function]}
  onLeftActionDeactivate={[Function]}
  onLeftActionRelease={[Function]}
  onLeftButtonsActivate={[Function]}
  onLeftButtonsCloseComplete={[Function]}
  onLeftButtonsCloseRelease={[Function]}
  onLeftButtonsDeactivate={[Function]}
  onLeftButtonsOpenComplete={[Function]}
  onLeftButtonsOpenRelease={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onPanAnimatedValueRef={[Function]}
  onRef={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onRightActionActivate={[Function]}
  onRightActionComplete={[Function]}
  onRightActionDeactivate={[Function]}
  onRightActionRelease={[Function]}
  onRightButtonsActivate={[Function]}
  onRightButtonsCloseComplete={[Function]}
  onRightButtonsCloseRelease={[Function]}
  onRightButtonsDeactivate={[Function]}
  onRightButtonsOpenComplete={[Function]}
  onRightButtonsOpenRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeComplete={[Function]}
  onSwipeMove={[Function]}
  onSwipeRelease={[Function]}
  onSwipeStart={[Function]}
  rightActionActivationDistance={125}
  rightActionReleaseAnimationConfig={null}
  rightActionReleaseAnimationFn={null}
  rightButtonWidth={85}
  rightButtonsActivationDistance={75}
  rightButtonsCloseReleaseAnimationConfig={null}
  rightButtonsCloseReleaseAnimationFn={null}
  rightButtonsOpenReleaseAnimationConfig={null}
  rightButtonsOpenReleaseAnimationFn={null}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  swipeReleaseAnimationConfig={
    Object {
      "duration": 250,
      "easing": [Function],
      "toValue": Object {
        "x": 0,
        "y": 0,
      },
    }
  }
  swipeReleaseAnimationFn={[Function]}
  swipeStartMinDistance={15}
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="link-wrapper"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "paddingLeft": 22,
              "paddingRight": 11,
              "paddingVertical": 11,
            },
          ]
        }
        testID="main-wrapper"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "width": 40,
              }
            }
            testID="avatar-button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EDF1F5",
                    "borderColor": "#DBE0E3",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1C2530",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "600",
                      "textAlign": "left",
                    },
                  ]
                }
                testID="text"
                weight="semiBold"
              >
                SJ
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingLeft": 11,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1C2530",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "textAlign": "left",
                    },
                  ]
                }
                testID="heading"
                variant="h4"
              >
                Stephen John
              </Text>
            </View>
            <Text
              alignment="right"
              muted={true}
              style={
                Array [
                  Object {
                    "color": "#97A4AB",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "textAlign": "right",
                  },
                ]
              }
              testID="user-id"
              variant="subtitle2"
            >
              ID:512345
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`UserItem Rendering should render properly 1`] = `
<View
  leftActionActivationDistance={0}
  leftActionReleaseAnimationConfig={null}
  leftActionReleaseAnimationFn={null}
  leftButtonWidth={85}
  leftButtonsActivationDistance={75}
  leftButtonsCloseReleaseAnimationConfig={null}
  leftButtonsCloseReleaseAnimationFn={null}
  leftButtonsOpenReleaseAnimationConfig={null}
  leftButtonsOpenReleaseAnimationFn={null}
  onLayout={[Function]}
  onLeftActionActivate={[Function]}
  onLeftActionComplete={[Function]}
  onLeftActionDeactivate={[Function]}
  onLeftActionRelease={[Function]}
  onLeftButtonsActivate={[Function]}
  onLeftButtonsCloseComplete={[Function]}
  onLeftButtonsCloseRelease={[Function]}
  onLeftButtonsDeactivate={[Function]}
  onLeftButtonsOpenComplete={[Function]}
  onLeftButtonsOpenRelease={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onPanAnimatedValueRef={[Function]}
  onRef={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onRightActionActivate={[Function]}
  onRightActionComplete={[Function]}
  onRightActionDeactivate={[Function]}
  onRightActionRelease={[Function]}
  onRightButtonsActivate={[Function]}
  onRightButtonsCloseComplete={[Function]}
  onRightButtonsCloseRelease={[Function]}
  onRightButtonsDeactivate={[Function]}
  onRightButtonsOpenComplete={[Function]}
  onRightButtonsOpenRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeComplete={[Function]}
  onSwipeMove={[Function]}
  onSwipeRelease={[Function]}
  onSwipeStart={[Function]}
  rightActionActivationDistance={125}
  rightActionReleaseAnimationConfig={null}
  rightActionReleaseAnimationFn={null}
  rightButtonWidth={85}
  rightButtonsActivationDistance={75}
  rightButtonsCloseReleaseAnimationConfig={null}
  rightButtonsCloseReleaseAnimationFn={null}
  rightButtonsOpenReleaseAnimationConfig={null}
  rightButtonsOpenReleaseAnimationFn={null}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  swipeReleaseAnimationConfig={
    Object {
      "duration": 250,
      "easing": [Function],
      "toValue": Object {
        "x": 0,
        "y": 0,
      },
    }
  }
  swipeReleaseAnimationFn={[Function]}
  swipeStartMinDistance={15}
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="link-wrapper"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "paddingLeft": 22,
              "paddingRight": 11,
              "paddingVertical": 11,
            },
          ]
        }
        testID="main-wrapper"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "width": 40,
              }
            }
            testID="avatar-button"
          >
            <Image
              source={
                Object {
                  "uri": "https://www.google.com",
                }
              }
              style={
                Array [
                  Object {
                    "borderColor": "#FFFFFF",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              testID="avatar-image"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingLeft": 11,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1C2530",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "textAlign": "left",
                    },
                  ]
                }
                testID="heading"
                variant="h4"
              >
                Stephen John
              </Text>
            </View>
            <Text
              alignment="right"
              muted={true}
              style={
                Array [
                  Object {
                    "color": "#97A4AB",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "textAlign": "right",
                  },
                ]
              }
              testID="user-id"
              variant="subtitle2"
            >
              ID:512345
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
